.gitignore

```diff
@@ -1,4 +1,107 @@
-*/.idea/workspace.xml
-*/.idea/tasks.xml
-/DropwizardHelloWorld/target/*
+
+# Created by https://www.gitignore.io/api/java,maven,macos,intellij
+
+### Intellij ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+.idea/
+
+## File-based project format:
+*.iws
+
+## Plugin-specific files:
+
+# IntelliJ
+/out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
+
+### Intellij Patch ###
+# Comment Reason: https://github.com/joeblau/gitignore.io/issues/186#issuecomment-215987721
+
+*.iml
```

Comment:
In addition to ignoring `.idea/` directory you also have to ignore ``*.iml`files. Those are IntelliJ's modules file descriptors. I don't really understand why Jetbrains guys suggest to keep IntelliJ-specific files... :pensive: